### DISCUSSION TITLE
Code interpreter, Assistants API (EnableCodeInterpreter = True) or SessionsPythonPlugin?

### DISCUSSION THREAD

**Discussion by panicoenlaxbox** [microsoft/semantic-kernel#9975] [Q&A] [answered]
Hi everyone,

I want to ask you about the right approach to achieve what I want to do.

I would like to offer the final user a playground with their data files (a bunch of .csv files) and some related .pdf files (one or two) to augment knowledge of the model (RAG).

I have tested manually in two different scenarios.

The first one is with Assistant API (activating file search and code interpreter). Everything goes well but I lose the capacity to work with .parquet files (they are not supported yet) and additionally, I have to respect the limits of the API (number of files, size, etc.).

The other one is using dynamic sessions in Container Apps and the plugin `SessionsPythonPlugin`. For now, I only have tested default dynamic sessions but I think with a custom docker image I will can install additional libraries for working with .parquet (pyarrow or fastparquet). Moreover, with this approach, I suppose I will allocate containers with more hardware resources and without restrictions. However, I lose things that I like about the Assistants API (automatic RAG, automatic and "clever" truncation of the history, persistent threads, etc).

Maybe I can combine both, Assistants API with `EnableFileSearch = True`, but `EnableCodeInterpreter = False`, this is `SessionsPythonPlugin` instead of native code interpreter tool. 

What do you think about it? Any recommendation?

Thank you so much.

**Comment by sphenry** [comment, marked as answer]
@panicoenlaxbox I think this is a good approach, treat you .parquet file handler as a separate plugin that the assistant can use. If the assistant is struggling to find the right plugin, try modifying the metadata to include negative examples, e.g. "do not use this tool for anything other than .parquet files"
